Calorimetry/python/jetResolution/histPool.py

```diff
@@ -0,0 +1,151 @@
+      outHistInfo.append(histInfoEntry)
+    return outHistInfo
+
+  def fillHist(self, histName, val, histSetName=''):
+    """Fill histogram.
+
+    Return True if val is successfully added to histogrm, otherwise return
+    False.
+    """
+    if histSetName not in self.histDict2D:
+      self.initNewSetOfHists(histSetName, self.histInfo)
+      jerFitHistInfo = self.createJERFitHistInfo()
+      fitHistSetName = histSetName + '/fit' if histSetName else 'fit'
+      self.initNewSetOfHists(fitHistSetName, jerFitHistInfo)
+
+    if histName in self.histDict2D[histSetName]:
+      self.histDict2D[histSetName][histName].Fill(val)
+      return True
+    else:
+      print("Can't find the hitogram: %s" % histName)
+      return False
+
+  def fillAllHists(self, vTuple, histSetName=''):
+    """Fill all histograms."""
+    cosTheta = abs(vTuple.quark1.Pz() / vTuple.quark1.P())
+    self.fillHist('cosTheta', cosTheta, histSetName)
+
+    totalTrueE = vTuple.totalTrueVisible + vTuple.totalTrueInvisible
+    self.fillHist('true_totalE', totalTrueE.E(), histSetName)
+
+    genJetSum = vTuple.genJets[0] + vTuple.genJets[1]
+    self.fillHist('gen_InvMass', genJetSum.M(), histSetName)
+    self.fillHist('gen_totalE', genJetSum.E(), histSetName)
+
+    recoJetSum = vTuple.recoJets[0] + vTuple.recoJets[1]
+    self.fillHist('reco_InvMass', recoJetSum.M(), histSetName)
+    self.fillHist('reco_totalE', recoJetSum.E(), histSetName)
+
+    for i in range(1, len(self.cosThetaRange)):
+      if cosTheta > self.cosThetaRange[i - 1] and cosTheta < self.cosThetaRange[i]:
+        binNumber = i - 1
+        fillVal = vTuple.recoJets[0].E() + vTuple.totalTrueInvisible.E()
+        fitHistSetName = histSetName + '/fit' if histSetName else 'fit'
+        self.fillHist('cosThetaBin' + str(binNumber), fillVal, fitHistSetName)
+
+  def makeJERPlot(self, histSetName=''):
+    """Create JER plot."""
+    if histSetName not in self.histDict2D:
+      return S_ERROR("Hist set with following name is not found: %s" % histSetName)
+
+    for i in range(1, len(self.cosThetaRange)):
+      binNumber = i - 1
+      fitHistSetName = histSetName + '/fit' if histSetName else 'fit'
+      histName = 'cosThetaBin' + str(binNumber)
+      hist = self.histDict2D[fitHistSetName][histName]
+      if not hist:
+        return S_ERROR("No hist found. SetName: %s; histName: %s" % (fitHistSetName, histName))
+      if hist.GetEntries() == 0:
+        return S_ERROR("Hist is empty. SetName: %s; histName: %s" % (fitHistSetName, histName))
+
+      #  print('Calculate RMS90. histName: %s; histTitle: %s, nEntries: %d, mean: %f; RMS: %f' % (hist.GetName(), hist.GetTitle(), hist.GetEntries(), hist.GetMean(), hist.GetRMS()))
+      rms90Data = calculateRMS90(hist)
+      resolution = rms90Data['rms90']
+      resolutionError = resolution / math.sqrt(rms90Data['totalCounts'])
+
+      self.histDict2D[histSetName]['JER'].SetBinContent(i, resolution)
```

Comment:
do we want to fill the actual jet energy resolution value instead of the relative value, aka JER/jet-energy or JER/Mean. Iwould rather do the later, in case the calibration of the jet energy would be off as additional effect.

This can also be done in post processing as well though